src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -839,7 +852,11 @@ class IndirectCallTransformer
             // special candidate helper and we need to use the new 'this'.
             GenTreeCall* call = compiler->gtCloneCandidateCall(origCall);
             call->gtArgs.GetThisArg()->SetEarlyNode(compiler->gtNewLclvNode(thisTemp, TYP_REF));
-            call->SetIsGuarded();
+
+            if (IsChainingSupported())
+            {
+                call->SetIsGuarded();
```

Comment:
I don't understand this change

Reply:
@AndyAyersMS it's a theoretical case. IsGuarded is only used for GDV chaining. Imagine we have:
```
obj.Call1();
obj.Call2();
```
and `Call1()` is expanded using 2 classes and chaining is disabled. `Call2()` is expanded with just 1 class - it will try to "chain" with previou `Call1` that doesn't support chaining.

Reply:
Are you sure? I think `IsGuarded` is currently just used to annotate the inlining tree and inline XML. It means that a callee was GDV and we now decided to test for it and inline it. So I think you can just call it always here.

The inline tree and XML will get confused when one call site expands to multiple candidates. We should think about how we want to express something like multi-guess GDV for those situations. But that can wait.

Reply:
You're right, I've reverted that change as it's not needed